src/structured-data/structured-data-tags.js

```diff
@@ -263,23 +263,32 @@ function generateLiveBlogPostingData(structuredData = {}, story = {}, publisherC
 }
 
 function getEmbedUrl(cards) {
-  let embedUrl = "";
+  const matchSubtypeWithUrl = {
+    "dailymotion-embed-script": "dailymotion-url",
+    "instagram": "instagram-url",
+    "facebook-video": "facebook-url",
+    "tweet": "tweet-url",
+    "vimeo-video": "vimeo-url",
+    "brightcove-video": "player-url"
+  };
 
-  // not using the return value of top level find
-  // coz we only need the embed url
-  // find is used for early exit
-  cards.find((card) => {
+  for (const card of cards) {
     const storyElements = get(card, ["story-elements"], []);
-    return storyElements.find((elem) => {
-      if (elem["embed-url"]) {
-        embedUrl = elem["embed-url"];
-        return true;
+    for (const elem of storyElements) {
+      if (elem.subtype && matchSubtypeWithUrl[elem.subtype]) {
+        const key = matchSubtypeWithUrl[elem.subtype];
+        if (elem.metadata && elem.metadata[key]) {
+          return elem.metadata[key];
+        }
+      } else if (elem.type === "youtube-video" && elem.subtype === null) {
```

Comment:
does this need to be an `else if`?

Reply:
Need not. Can be an if condition itself. Will change it.